packages/actor-rdf-resolve-quad-pattern-rdfjs-source/lib/RdfJsQuadSource.ts

```diff
@@ -59,19 +84,26 @@ export class RdfJsQuadSource implements IQuadSource {
       let i = 0;
       cardinality = await new Promise((resolve, reject) => {
         const matches = this.source.match(
-          RdfJsQuadSource.nullifyVariables(subject),
-          RdfJsQuadSource.nullifyVariables(predicate),
-          RdfJsQuadSource.nullifyVariables(object),
-          RdfJsQuadSource.nullifyVariables(graph),
+          RdfJsQuadSource.nullifyVariables(subject, quotedTripleFiltering),
+          RdfJsQuadSource.nullifyVariables(predicate, quotedTripleFiltering),
+          RdfJsQuadSource.nullifyVariables(object, quotedTripleFiltering),
+          RdfJsQuadSource.nullifyVariables(graph, quotedTripleFiltering),
         );
         matches.on('error', reject);
         matches.on('end', () => resolve(i));
         matches.on('data', () => i++);
       });
     }
+
+    // If `match` would require filtering afterwards, our count will be an over-estimate.
+    const pattern = DF.quad(subject, predicate, object, graph);
+    const wouldRequirePostFiltering = (!quotedTripleFiltering &&
```

Comment:
Could we instead of issuing a separate `source.match()` in order to count pass in args a stream of processed results from our `match()`?

Atm we only pass an iterator, and it's being exhausted on read from it, so we can't use it to count.
Could we craft another readable stream in `match()` and filter results in there? Then we can both wrap it with `wrapAsyncIterator` and handle it over to user and pass it in here, and read it in order to count quads.

That would spare us the need to `source.match()` twice and would allow to provide `exact` estimate.
One concern is that we have `autoStart`: `false` on iterator, I guess that's for a reason, and doing this strategy does start reading in order to count, so I guess results would be eagerly resolved and available in the iterator. Don't know what adverse effects that might have.